### PR TITLE
Provide diagnostic for endpoints for XML DSL on save (#30)

### PR DESCRIPTION
**Pull request by apupier**
image

image

requires https://github.com/camel-tooling/camel-tooling-common/pull/26 to be reviewed and merged

### CODE REVIEW

File: src/main/java/com/github/cameltooling/lsp/internal/diagnostic/DiagnosticService.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+		Map<CamelEndpointDetailsWrapper, EndpointValidationResult> endpointErrors = computeCamelErrors(params);
+		List<Diagnostic> diagnostics = converToLSPDiagnostics(endpointErrors, camelLanguageServer.getTextDocumentService().getOpenedDocument(params.getTextDocument().getUri()));
+		PublishDiagnosticsParams diagnosticParam = new PublishDiagnosticsParams(params.getTextDocument().getUri(), diagnostics);
+		camelLanguageServer.getClient().publishDiagnostics(diagnosticParam);
+	}
+
+	private Map<CamelEndpointDetailsWrapper, EndpointValidationResult> computeCamelErrors(DidSaveTextDocumentParams params) {
+		Map<CamelEndpointDetailsWrapper, EndpointValidationResult> endpointErrors = new HashMap<>();
+
+		if (params.getTextDocument().getUri().endsWith(".xml")) {
```
Comment (lhein) [inline src/main/java/com/github/cameltooling/lsp/internal/diagnostic/DiagnosticService.java]: does that mean we only support validation on xml so far?

**Comment by lhein** [comment]
why do we need roaster?

File: src/main/java/com/github/cameltooling/lsp/internal/diagnostic/DiagnosticService.java (hunk 1 above)
Comment (apupier) [reply src/main/java/com/github/cameltooling/lsp/internal/diagnostic/DiagnosticService.java]: yes.
I created this epic to gather all other closely related feature that would be nice to support soon: https://github.com/camel-tooling/camel-language-server/issues/111

**Comment by apupier** [comment]
> why do we need roaster?

good point.
It seems that there is a problem in camel-route-parser dependency, it is using it but not declaring it as a dependency. I have a ClassNotFoundException on roaster classes if I don't add it to this pom

**Comment by lhein** [comment]
Roaster is usually used for parsing java if I understand correctly.

**Comment by apupier** [comment]
Roaster is used to parse both xml and java

only the roaster specific to java is declared (as provided :'( )
https://github.com/apache/camel/blob/a07781c240fa81d500ab13b6f4a85bb7d20daf28/tooling/camel-route-parser/pom.xml#L37-L43

**Comment by apupier** [comment]
merging as approved but we can of course continue discussion on scope/impact/details of this PR

**Pull request merged**